server/src/main/java/org/elasticsearch/action/support/local/LocalClusterStateRequest.java

```diff
@@ -49,8 +49,12 @@ protected LocalClusterStateRequest(StreamInput in) throws IOException {
         in.readBoolean();
     }
 
+    /**
+     * Most transport actions that run on the local node will never need to serialize their request. The exceptions are actions that are
+     * run across clusters and will thus need to serialize requests for communication between the clusters.
+     */
     @Override
-    public final void writeTo(StreamOutput out) throws IOException {
+    public void writeTo(StreamOutput out) throws IOException {
```

Comment:
Yeah I don't really like removing this `final`. I think I'd rather we just made `TransportGetIndexAction extends HandledTransportAction` rather than trying to generalize `TransportClusterInfoAction` for this one special case.

Reply:
Hmm that means we'd have to re-implement/copy the code to wait for cluster blocks. Also, the `TransportClusterStateAction` faces the same issue -- and that one also needs to wait for the master in the cluster state.
We can try extracting that waiting logic so `TransportLocalClusterStateAction`, `TransportGetIndexAction`, and `TransportClusterStateAction` can share it, but that might get a bit ugly.